sdk/go/common/slice/slice.go

```diff
@@ -23,3 +23,27 @@ func Prealloc[T any](capacity int) []T {
 	}
 	return make([]T, 0, capacity)
 }
+
+// Map applies the given function to each element of the given slice and returns a new slice with the results.
+func Map[T, U any](s []T, f func(T) U) []U {
+	r := make([]U, len(s))
```

Comment:
```suggestion
	r := Prealloc[U](len(s))
```

Is there  a reason not to use the Prealloc function here?